airbyte-integrations/connectors/source-typeform/source_typeform/source.py

```diff
@@ -0,0 +1,230 @@
+    token_field: str = "token"
+
+    @property
+    def limit(self):
+        return super().limit
+
+    state_checkpoint_interval = limit
+
+    @abstractmethod
+    def get_updated_state(
+        self,
+        current_stream_state: MutableMapping[str, Any],
+        latest_record: Mapping[str, Any],
+    ) -> Mapping[str, Any]:
+        pass
+
+    def next_page_token(self, response: requests.Response) -> Optional[str]:
+        items = response.json()["items"]
+        if items and len(items) == self.limit:
+            return items[-1][self.token_field]
+        return None
+
+
+class Responses(StreamMixin, IncrementalTypeformStream):
+    """
+    This stream is responsible for fetching responses for particulat form_id.
+    API doc: https://developer.typeform.com/responses/reference/retrieve-responses/
+    """
+
+    primary_key = "response_id"
+    limit: int = 1000
+
+    def path(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> str:
+        return f"/forms/{stream_slice['form_id']}/responses"
+
+    def get_form_id(self, record: Mapping[str, Any]) -> Optional[str]:
+        """
+        Fetches form id to which current record belongs.
+        """
+        referer = record.get("metadata", {}).get("referer")
+        return referer.rsplit("/")[-1] if referer else None
+
+    def get_updated_state(
+        self,
+        current_stream_state: MutableMapping[str, Any],
+        latest_record: Mapping[str, Any],
+    ) -> Mapping[str, Any]:
+        form_id = self.get_form_id(latest_record)
+        if not form_id or not latest_record.get(self.cursor_field):
+            return current_stream_state
+
+        current_stream_state[form_id] = current_stream_state.get(form_id, {})
+        current_stream_state[form_id][self.cursor_field] = max(
+            pendulum.from_format(latest_record[self.cursor_field], self.date_format).int_timestamp,
+            current_stream_state[form_id].get(self.cursor_field, 1),
+        )
+        return current_stream_state
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        params = {"page_size": self.limit}
+        stream_state = stream_state or {}
+
+        if not next_page_token:
+            # use state for first request in incremental sync
+            params["sort"] = "submitted_at,asc"
+            # start from last state or from start date
+            since = max(self.start_date.int_timestamp, stream_state.get(stream_slice["form_id"], {}).get(self.cursor_field, 1))
+            if since:
+                params["since"] = pendulum.from_timestamp(since).format(self.date_format)
+        else:
+            # use response token for pagination after first request
+            # this approach allow to avoid data duplication whithin single sync
+            params["after"] = next_page_token
```

Comment:
According to https://developer.typeform.com/responses/reference/retrieve-responses/

if we use 'after' then it  "sorts responses in the order that our system processed them (submitted_at). "

does it mean that the order is the same as in the initial request (by default it )?  Default value is submitted_at,desc

Reply:
`after` returns data in ascending order by `submitted_at` field